src/sentry/models/actor.py

```diff
@@ -23,6 +23,40 @@ def actor_type_to_class(type: int) -> Type[Union["Team", "User"]]:
     return ACTOR_TYPE_TO_CLASS[type]
 
 
+def fetch_actor_by_actor_id(cls, actor_id: int) -> Type[Union["Team", "User"]]:
+    from sentry.models import Team, User
+
+    def team_fetcher(actor_id: int):
+        return Team.objects.get(actor_id=actor_id)
+
+    def user_fetcher(actor_id: int):
+        from sentry.services.hybrid_cloud.user import user_service
+
+        user = user_service.get_by_actor_id(actor_id)
+        if user is None:
+            raise User.DoesNotExist
+        return user
```

Comment:
For instance, this could just be a `User.objects.get(user.id)`.  That would allow our upstream type to be anything, and we can still later decide to down or up grade this code path based on how we chose to handle the caching layer.